compiler/damlc/daml-stdlib-src/DA/Internal/Template/Functions.daml

```diff
@@ -58,6 +58,19 @@ class HasFetch t where
   -- contract, this fails and aborts the entire transaction.
   fetch : ContractId t -> Update t
 
+-- | Exposes `softFetch` function
+class HasSoftFetch t where
+  -- | HIDE
+  _softFetch : ContractId t -> Update t
+
+#ifdef DAML_PACKAGE_UPGRADES
+
+-- | TODO
+softFetch : HasSoftFetch t => ContractId t -> Update t
+softFetch = _softFetch
```

Comment:
Why indirect here? (We don't do this for : `HasFetch` / `fetch`)

Reply:
good question, the reason i went for the split approach is that the GHC fork doesn't know about LF versions, so when we start generating `HasSoftFetch` instances for templates (in this PR, these are written manually in the source `.daml`), the class, including the method `_softFetch` will need to be visible (otherwise we'd get "unknown typeclass HasSoftFetch" errors). To prevent that from being used, we can still hide the _function_ `softFetch` behind CPP directives